src/services/PizzaImagery/PizzaImagery.js

```diff
@@ -0,0 +1,52 @@
+import {API} from '../API';
+import {Auth} from '../Auth';
+
+/**
+ * Supplies pizza-related images
+ */
+class PizzaImageryService {
+  constructor() {
+    this.images = {};
+    console.log('PizzaImageryService constructor');
+    this.loadIngredients().then(data => {
+      data.push({
+        name: 'pizza',
+        url: 'static/images/pizza.png',
+      });
+      return data;
+    }).then(list => {
+      console.log('PizzaImageryService constructor', list);
+      list.forEach(item => {
+        const image = new Image();
+        image.crossOrigin = '';
+        image.src = API.prependAssetUrl(item.url);
+        this.images[item.name] = Promise.resolve(image);
```

Comment:
I didn't understand why you need to store consumed Promises in hash table. Since you don't have any async code after.